pennylane/gradients/vjp.py

```diff
@@ -363,22 +364,35 @@ def func(_, num=None):  # pylint: disable=unused-argument
     except (AttributeError, TypeError, NotImplementedError):
         pass
 
-    gradient_tapes, fn = gradient_fn(tape, **gradient_kwargs)
+    gradient_tapes, fn = gradient_fn(tape, shots=shots, **gradient_kwargs)
 
     def processing_fn(results, num=None):
         # postprocess results to compute the Jacobian
         jac = fn(results)
+        shot_vector = isinstance(shots, Sequence)
+
         if qml.active_return():
             multi = len(tape.measurements) > 1
+
+            if not shot_vector:
+                if multi:
+                    return compute_vjp_multi_new(dy, jac, num=num)
+                return compute_vjp_single_new(dy, jac, num=num)
+
             if multi:
-                return compute_vjp_multi_new(dy, jac, num=num)
-            return compute_vjp_single_new(dy, jac, num=num)
+                vjp_ = [compute_vjp_multi_new(dy_, jac_, num=num) for dy_, jac_ in zip(dy, jac)]
+            else:
+                vjp_ = [compute_vjp_single_new(dy_, jac_, num=num) for dy_, jac_ in zip(dy, jac)]
+
+            return qml.math.sum(qml.math.stack(vjp_), 0)

```

Comment:
Why do you need to introduce the sum here?

Reply:
The sum is over the gradients for the different shot entries. We do this because autograd doesn't allow the gradient of each parameter to be a tuple. I think this is a similar case to summing over the gradients for the different measurements as well.

Reply:
Thank you !